x-pack/plugins/code/server/routes/file.ts

```diff
@@ -149,7 +149,7 @@ export function fileRoute(server: hapi.Server, options: ServerOptions) {
     const after = queries.after !== undefined;
     try {
       const repository = await gitOperations.openRepo(uri);
-      const commit = await gitOperations.getCommit(repository, ref);
+      const commit = await gitOperations.getCommit(repository, decodeURIComponent(ref));
```

Comment:
@spacedragon this will raise exception if pass through a branch with `/`.  try this repo: https://github.com/elastic/apm-agent-nodejs/tree/greenkeeper/ws-6.2.1

Reply:
![image](https://user-images.githubusercontent.com/987855/55702022-c9e4d980-5a07-11e9-966d-cb071be0798d.png)

both the commented code (borrowed from [here](https://github.com/nodegit/nodegit/issues/1370)) and uncommented code in the screenshot above work.

I think the `getCommit(xxx)` function needs to be more comprehensive to cover all cases (branch/tag/remote, etc) of revisions. https://stackoverflow.com/a/50192006

let's merge this PR first and use a separate issue to track this.